src/core/toxid.cpp

```diff
@@ -126,12 +123,10 @@ ToxId::ToxId(const uint8_t* rawId, int len)
 
 void ToxId::constructToxId(const QByteArray& rawId)
 {
-    // TODO: remove construction from PK only
-    if (rawId.length() == TOX_PUBLIC_KEY_SIZE) {
-        toxId = QByteArray(rawId); // construct from PK only
-    } else if (rawId.length() == TOX_ADDRESS_SIZE && isToxId(rawId.toHex().toUpper())) {
+    if (rawId.length() == TOX_ADDRESS_SIZE && isToxId(rawId.toHex().toUpper())) {
         toxId = QByteArray(rawId); // construct from full toxid
     } else {
+        assert(!"ToxId constructed with invalid input");
```

Comment:
Out of scope for this, but something I like doing is having a static create method that returns a `std::optional<MyType>` when I have a fallible constructor.

Reply:
Agreed, ToxId almost has that with the (private) `void ToxId::constructToxId(const QByteArray& rawId)` but it still can't return null.

I'd also like to remove the `isValid` from ToxId and just have it fail construction if not valid, but will leave those for later.

Reply:
For future reference, this seems like a no-go with Qt. Any class used in signals/slots must have a public default constructor, so a strict factory pattern isn't workable with ToxPk which we want to be able to pass around easily. This also means keeping the `isValid` is basically required, and the recommended "Qt" way of doing things. Oh well =/